### PR TITLE
Fix #19451 (spurious incompatible-prefix warning when using "as pattern")

### LINKED ISSUE #19451: Broken Link in "list of math projects"
# Description of the problem

Broken link for the "coq user contributions repository" page in [wiki](https://github.com/coq/coq/wiki/List%20of%20Coq%20Math%20Projects) of the "list of math projects".

### LINKED ISSUE #19541: Incorrect warning about notation conflict between "term" and "term as pattern"
### Description of the problem

The code shown below produces this warning:
```
Warning: Notations "{[ _ ]}" defined at level 1 with arguments constr at level 200 and "{[ _ | _ ]}"
defined at level 1 with arguments constr  as pattern at level 200 have incompatible prefixes.
One of them will likely not work. [notation-incompatible-prefix,parsing,default]
```
However, as far as I understand the explanation given by Thibaut Pérami [over here](https://gitlab.mpi-sws.org/iris/stdpp/-/issues/216#note_104616), this should not be a grammar conflict: `as pattern` is supposed to mean "parse like a term, but then convert the term to a pattern later". And indeed both notations work just fine together.

So, I think the warning is wrong and should not be shown in this case. Or am I misunderstanding something?

### Small Coq file to reproduce the bug

```coq
Class Singleton A B := singleton: A -> B.
Notation "{[ x ]}" := (singleton x) (at level 1, x at level 200).

Record propset (A : Type) : Type := PropSet { propset_car : A -> Prop }.
Notation "{[ x | P ]}" := (PropSet (fun x => P))
  (at level 1, x at level 200 as pattern, format "{[  x  |  P  ]}").
```

### Version of Coq where this bug occurs

8.20.0

### Last version of Coq where the bug did not occur

_No response_

### PR DESCRIPTION
**Pull request by proux01**
Fixes / closes #19541 

- [x] Added / updated **test-suite**.

- [x] Added **changelog**.
- ~[ ] Added / updated **documentation**.~
  
  - ~[ ] Documented any new / changed **user messages**.~
  - ~[ ] Updated **documented syntax** by running `make doc_gram_rsts`.~

- ~[ ] Opened **overlay** pull requests.~

<!-- Pointers to relevant developer documentation:

Contributing guide: https://github.com/coq/coq/blob/master/CONTRIBUTING.md

Test-suite: https://github.com/coq/coq/blob/master/test-suite/README.md

Changelog: https://github.com/coq/coq/blob/master/doc/changelog/README.md

Building the doc: https://github.com/coq/coq/blob/master/doc/README.md
Sphinx: https://github.com/coq/coq/blob/master/doc/sphinx/README.rst
doc_gram: https://github.com/coq/coq/blob/master/doc/tools/docgram/README.md

Overlays: https://github.com/coq/coq/blob/master/dev/ci/user-overlays/README.md

### CODE REVIEW

**Comment by proux01** [comment]
@herbelin would you be the assignee for this trivial fix?

File: parsing/extend.ml [hunk 1]
```diff
@@ -37,17 +37,21 @@ type 'a constr_entry_key_gen =
...
   | ETGlobal, ETGlobal -> true
   | ETBigint, ETBigint -> true
   | ETBinder b1, ETBinder b2 -> b1 == b2
   | ETConstr (s1,bko1,_lev1), ETConstr (s2,bko2,_lev2) ->
-    Notationextern.notation_entry_eq s1 s2 && Option.equal (=) bko1 bko2
+    Notationextern.notation_entry_eq s1 s2 && binder_kind_eq bko1 bko2
   | ETPattern (b1,n1), ETPattern (b2,n2) -> b1 = b2 && Option.equal Int.equal n1 n2
   | (ETIdent | ETName | ETGlobal | ETBigint | ETBinder _ | ETConstr _ | ETPattern _), _ -> false
 
+let constr_entry_key_eq = constr_entry_key_eq_gen (Option.equal (=))
```
Comment (ppedrot) [inline parsing/extend.ml:51]: While we're at it I think you could remove the call to the generic equality. It's a footgun waiting to blow up.

Comment (proux01) [reply parsing/extend.ml]: Indeed, done

**Review by herbelin — APPROVED** [review]
Looks good.

**Comment by proux01** [comment]
@coqbot run full ci

**Comment by proux01** [comment]
@herbelin anything preventing the merge of this trivial fix?

**Comment by herbelin** [comment]
@coqbot merge now

**Comment by proux01** [comment]
Thanks

**Pull request merged**